### PR TITLE
Support turning off the model uploading in ClearML

### LINKED ISSUE #20889: Disable ClearML automatic model upload
### Feature request

Allowing users of `transformers` to disable the automatic model uploading of ClearML. Perhaps we can also allow users to write their own integration callbacks in case they want to configure some more stuff.

The place where the saving happens is `src/transformers/integrations.py`:
```
    def on_save(self, args, state, control, **kwargs):
        if self._clearml_task and state.is_world_process_zero:
            ckpt_dir = f"checkpoint-{state.global_step}"
            artifact_path = os.path.join(args.output_dir, ckpt_dir)
            logger.info(f"Logging checkpoint artifacts in {ckpt_dir}. This may take time.")
            self._clearml_task.update_output_model(artifact_path, iteration=state.global_step, auto_delete_file=False)
```

We should also add a condition to the main `if`, similar to what `NeptuneCallback`  or `MLflowCallback` is doing.

### Motivation

Several experiments that I ran were interrupted because I reached the max limit of model uploading. However, I was not interested in uploading my models in the first place. Hence, a configuration would be appropriate in this case.

### Your contribution

I can submit a PR if the contributors would help figure out the correct way of handling it :)

### PR DESCRIPTION
**Pull request by david1542**
# What does this PR do?
Support turning off the model uploading in ClearML using an environment variable called `CLEARML_LOG_MODEL`

Fixes #20889 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! Could you just make sure to document this new env variable in the docstring of the callback (along the other ones)?

**Comment by dudulasry** [comment]
Hi @sgugger :) I modified the docstring, as you suggested. Can you please have a look? 🙏

**Comment by sgugger** [comment]
Code looks good but it seems there is an issue with your CircleCI permissions, the tests won't run.
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

**Comment by david1542** [comment]
@sgugger I re-authenticated to CircleCI, and it seems that the CI passed :)
Can you have a look and approve? :)

File: src/transformers/integrations.py [hunk 1]
```diff
@@ -1313,6 +1313,8 @@ class ClearMLCallback(TrainerCallback):
         ClearML project name.
     - **CLEARML_TASK** (`str`, *optional*, defaults to `Trainer`):
         ClearML task name.
+            CLEARML_LOG_MODEL (`bool`, *optional*, defaults to `False`):
+                Whether to log models as artifacts during training.
```
Comment (sgugger) [inline src/transformers/integrations.py:1317]: As you can see above we just merged a PR that slightly changed the doc of this class. Could you adapt your addition?
```suggestion
    - **CLEARML_LOG_MODEL** (`bool`, *optional*, defaults to `False`):
        Whether to log models as artifacts during training.
```

**Review by sgugger — APPROVED** [review]
Thanks a lot! Just one last comment and we should be good to merge.

**Comment by david1542** [comment]
@sgugger Thanks for the feedback. I accepted your change :)

**Comment by sgugger** [comment]
Thanks for your contribution!

**Pull request merged**